apps/web/src/pages/Communities/CommunityMembersPage/components/EditCommunityMemberDialog.tsx

```diff
@@ -0,0 +1,196 @@
+} from "@gc-digital-talent/graphql";
+
+import { getFullNameLabel } from "~/utils/nameUtils";
+import { CommunityMember } from "~/utils/communityUtils";
+
+import { CommunityMemberFormValues, ContextType } from "./types";
+import { getTeamBasedRoleOptions } from "./utils";
+import useAvailableRoles from "./useAvailableRoles";
+import { UpdateUserCommunityRoles_Mutation } from "./operations";
+
+interface EditCommunityMemberDialogProps {
+  user: CommunityMember;
+  community: CommunityMembersPageCommunityFragmentType;
+}
+
+const EditCommunityMemberDialog = ({
+  user,
+  community,
+}: EditCommunityMemberDialogProps) => {
+  const intl = useIntl();
+  const { teamId } = useOutletContext<ContextType>();
+  const { roles, fetching } = useAvailableRoles();
+  const [, executeMutation] = useMutation(UpdateUserCommunityRoles_Mutation);
+  const [isOpen, setIsOpen] = useState<boolean>(false);
+  const initialRolesIds = user.roles.map((role) => role.id);
+
+  const methods = useForm<CommunityMemberFormValues>({
+    defaultValues: {
+      userId: user.id,
+      userDisplay: user.id,
+      communityId: community.id,
+      roles: initialRolesIds,
+    },
+  });
+
+  const {
+    handleSubmit,
+    formState: { isSubmitting },
+  } = methods;
+
+  const handleSave = async (formValues: CommunityMemberFormValues) => {
+    const rolesToAttach = formValues.roles.filter(
+      (role) => !initialRolesIds.includes(role),
+    );
+    const rolesToAttachArray: RoleInput[] = rolesToAttach.map((role) => {
+      return { roleId: role, teamId };
+    });
+    const rolesToDetach = initialRolesIds.filter(
+      (role) => !formValues.roles.includes(role),
+    );
+    const rolesToDetachArray: RoleInput[] = rolesToDetach.map((role) => {
+      return { roleId: role, teamId };
+    });
+
+    await executeMutation({
+      updateUserRolesInput: {
+        userId: formValues.userId,
+        roleAssignmentsInput: {
+          attach: rolesToAttachArray.length ? rolesToAttachArray : undefined,
+          detach: rolesToDetachArray.length ? rolesToDetachArray : undefined,
+        },
+      },
+    })
+      .then((res) => {
+        if (!res.error) {
+          setIsOpen(false);
+          toast.success(
+            intl.formatMessage({
+              defaultMessage: "Roles updated successfully",
+              id: "Jc9FDx",
+              description:
+                "Alert displayed to user when a community member's roles have been updated",
+            }),
+          );
+        }
+      })
+      .catch(() => {
+        toast.error(
+          intl.formatMessage({
+            defaultMessage: "Role update failed",
+            id: "bmOZXl",
+            description:
+              "Alert displayed to user when an error occurs while editing a community member's roles",
+          }),
+        );
+      });
+  };
+
+  const roleOptions = getTeamBasedRoleOptions(roles, intl);
```

Comment:
Same thing here about filtering out admin if you don't have the permission to assign that role.

Reply:
d619b1a0c305ce96eb85e4c14c27ebc1836ccadd